File: src/main/java/com/raizlabs/android/dbflow/structure/SQLiteType.java
package com.raizlabs.android.dbflow.structure;

import java.util.HashMap;

/**
 * Author: andrewgrosner
 * Contributors: { }
 * Description:
 */
public enum SQLiteType {

    INTEGER,
    REAL,
    TEXT,
    BLOB;

    private static final HashMap<Class, SQLiteType> sTypeMap = new HashMap<Class, SQLiteType>() {
        {
            put(byte.class, SQLiteType.INTEGER);
            put(short.class, SQLiteType.INTEGER);
            put(int.class, SQLiteType.INTEGER);
            put(long.class, SQLiteType.INTEGER);
            put(float.class, SQLiteType.REAL);
            put(double.class, SQLiteType.REAL);
            put(boolean.class, SQLiteType.INTEGER);
            put(char.class, SQLiteType.TEXT);
            put(byte[].class, SQLiteType.BLOB);
            put(Byte.class, SQLiteType.INTEGER);
            put(Short.class, SQLiteType.INTEGER);
            put(Integer.class, SQLiteType.INTEGER);
            put(Long.class, SQLiteType.INTEGER);
            put(Float.class, SQLiteType.REAL);
            put(Double.class, SQLiteType.REAL);
            put(Boolean.class, SQLiteType.INTEGER);
            put(Character.class, SQLiteType.TEXT);
            put(String.class, SQLiteType.TEXT);
            put(Byte[].class, SQLiteType.BLOB);
        }
    };

    public static boolean containsClass(Class clazz) {
        return sTypeMap.containsKey(clazz);
    }

    public static SQLiteType get(Class clazz) {
        return sTypeMap.get(clazz);
    }
}

File: dbflow-core-common/src/main/kotlin/com/raizlabs/dbflow5/sql/SQLiteType.kt
package com.raizlabs.dbflow5.sql

/**
 * Description: Represents a type that SQLite understands.
 */
enum class SQLiteType {

    /**
     * Represents an integer number in the DB.
     */
    INTEGER,

    /**
     * Represents a floating-point, precise number.
     */
    REAL,

    /**
     * Represents text.
     */
    TEXT,

    /**
     * A column defined by [byte[]] data. Usually reserved for images or larger pieces of content.
     */
    BLOB;


    companion object {

        private val typeMap = platformTypeMap

        /**
         * Returns the [SQLiteType] for this class
         *
         * @param className The fully qualified class name
         * @return The type from the class name
         */

        operator fun get(className: String): SQLiteType? = typeMap[className]

        fun containsClass(className: String): Boolean = typeMap.containsKey(className)
    }
}

/**
 * The type map that we add to the contained type map specific to a platform.
 */
expect val platformTypeMap: Map<String, SQLiteType>
